Tribler/community/doubleentry/community.py

```diff
@@ -89,154 +75,73 @@ def publish_signature_request_message(self, candidate):
-                            destination=(signature_request.candidate,),
-                            # Append the signature to the data
-                            payload=data_without_signature + (signature,))
+        
+        payload = (payload.up, payload.down, 
+                   payload.total_up, payload.total_down, 
+                   payload.sequence_number_requester, payload.previous_hash_requester,
+                   sequence_number_responder, previous_hash_responder)
+        
+        meta = self.get_meta_message(SIGNATURE)
+        message = meta.impl(authentication=(message.authentication.members,),
+                            distribution=(message.distribution.global_time,),
+                            payload=payload)
         return message
+    
+    
+    def allow_signature_response(self, request, response, modified):
+        """
+        We've received a signature repsonse message, we must return either:
+            a. True, if we accept this message
+            b. False, if not (because of inconsistencies in the payload)
+        """
+        
+        if request.payload.sequence_number_requester == response.payload.sequence_number_requester and \
+            request.payload.previous_hash_requester == response.payload.previous_hash_requester:
+            
+            #our values did not change, accept
+            self.persist_signature_response(response)
+            return True
+        return False
 
-    def check_signature_response(self, messages):
-        """
-        Generator that generates a list of correct signature_response messages from a list of messages.
-        :param messages: Potential valid signature_response messages.
-        :return: Series of correct signature_response messages
-        """
-        self._logger.info("Received " + str(len(messages)) + " signature responses.")
-        for message in messages:
-            # Check if it is not from ourselves.
-            if message.payload.public_key_responder != self._public_key:
-                payload = message.payload
-                # Check if the request part is valid
-                valid_request = self.validate_signature(
-                    payload.public_key_requester, payload.signature_data_requester(), payload.signature_requester)
-                # Check if the response part is valid
-                valid_response = self.validate_signature(
-                    payload.public_key_responder, payload.signature_data_responder(), payload.signature_responder)
-                if valid_request and valid_response:
-                    self._logger.info("Received valid response.")
-                    yield message
-                else:
-                    yield DropMessage("Invalid signature response message.")
-            else:
-                yield DropMessage(message, "Invalid signature response message, send by me.")
-
-    def on_signature_response(self, messages):
```

Comment:
Because the message is modified by the responder, the responder itself does not have the double signed message. Maybe we should send it to him from within the dispersy core.